source/s3_client.c

```diff
@@ -1763,6 +1815,17 @@ static bool s_s3_client_should_update_meta_request(
         }
     }
 
+    if (meta_request->fio_opts.streaming_upload) {
```

Comment:
can you add more doc somehow of how high level fio streaming reqs are scheduled. 
i.e. it will not schedule any more reqs if there are too many reqs in flight. 
what happens if we have fio streaming req along with other reqs? other reqs will keep pipeline full and num reqs in flight will never go down? does that mean that fio streaming does not get scheduled?